### ISSUE TITLE
Edge color in svg graphs

### ISSUE THREAD

**Issue by wrick17** [jacomyal/sigma.js#575]
Hi,

When I dynamically upadate the edge.color property, the color changes fine with the canvas and webgl renderer. But the color does not update when I am using the svg renderer. Any suggestions?

**Comment by Yomguithereal** [comment]
Hello @wrick17. This is probably because I forgot to deal with colors in the dedicated renderer. Probably because my use cases were dealing with a `freeStyle` renderer so that CSS classes can apply etc. Let me check that.

**Comment by wrick17** [comment]
Thanks. Well I am new to this, so i am using the default one only, thats how i came to see the issue.

**Comment by sim51** [comment]
With freestyle to true or false, CSS class are applied.
In the 'svg-freestyke-renderer.html, if I remove css, and put a red color on edge, it's working.

**Comment by wrick17** [comment]
I am trying to use the default svg renderer. there it is not applying.

**Comment by sim51** [comment]
What is for you the default SVG renderer  (for me I use it) ? 
Can you create a plunker with an example of your code ?

**Comment by wrick17** [comment]
err posting the code is a bit tough. its way too complex. i'll post a snippet below:

$.each(s.graph.nodes(), function(index, node) {
      if (mod_array[node.attributes.modularityclass] === undefined) {
        node.color = getRandomColor();
        mod_array[node.attributes.modularityclass] = node.color;
      } else {
        node.color = mod_array[node.attributes.modularityclass];
      }
    });

    $.each(s.graph.edges(), function(index, edge) {
      edge.color = s.graph.nodes()[edge.source.split('').splice(1, edge.source.split('').length - 1).join('')].color;
    });

so this is how i am adding color to the nodes and edges. so nodes and edges from the same community have the same color.  then i do this:

s.startForceAtlas2({
        strongGravityMode: true,
        gravity: 0.1,
        worker: true,
        scalingRatio: 1,
        slowDown: 2,
        barnesHutOptimize: true,
        barnesHutTheta: 0.1,
        nodesize: 'original'
      });

      s.refresh({
        skipIndexation: true
      });

and the graph renders. it goes well with canvas and webgl. but doesnt go for svg. 

PS: sorry for the bad formatting. i m really new to this. :p

**Comment by sim51** [comment]
I think the bug come from the `update` method into the file `sigma.svg.edges.def.js`.
If I add this line it's work `line.setAttributeNS(null, 'stroke', edge.color);`

**Comment by wrick17** [comment]
@sim51 where do i add it? a little detail would really help. thanks

**Comment by sim51** [comment]
https://github.com/jacomyal/sigma.js/blob/master/src/renderers/svg/sigma.svg.edges.def.js#L62

**Comment by wrick17** [comment]
@sim51 sorry. no luck. still taking the default color.
i tried to change it to this:

    line.setAttributeNS(null, 'stroke-width', edge[prefix + 'size'] || 1);
      line.setAttributeNS(null, 'stroke', edge.color);
      line.setAttributeNS(null, 'x1', source[prefix + 'x']);
      line.setAttributeNS(null, 'y1', source[prefix + 'y']);
      line.setAttributeNS(null, 'x2', target[prefix + 'x']);
      line.setAttributeNS(null, 'y2', target[prefix + 'y']);

still it takes the default color.

**Comment by sim51** [comment]
Look at this example : http://embed.plnkr.co/Rvgzjq5vor97lFKNl1vj/preview
All edges are blue, not red

**Comment by wrick17** [comment]
@sim51 so can i update the graph without recreating the sigma instance? i mean i am using forceAtlas2 which adds properties to the running instance. based on that i have to change the colors and update the graph.

**Comment by sim51** [comment]
@wrick17 yes of course, it what I do into the example ... I have created a sigma instance with red edge, and after (with your code snippet) I change all edge color to blue (and it's the same sigma instance!).
If you want to run a forceatlas after the color modification, you can. I have added it to the example.
